MQTTnet.Core/Adapter/MqttChannelCommunicationAdapter.cs

```diff
@@ -83,23 +83,27 @@ public MqttChannelCommunicationAdapter(IMqttCommunicationChannel channel, IMqttP
                 {
                     foreach (var packet in packets)
                     {
-                        MqttTrace.Information(nameof(MqttChannelCommunicationAdapter), "TX >>> {0} [Timeout={1}]", packet, timeout);
+                        if (packet == null) {continue; }
+
+                        MqttTrace.Information(nameof(MqttChannelCommunicationAdapter), $"TX >>> {packet} [Timeout={timeout}]");
```

Comment:
Please do not use string interpolation for logging. While optimizing performance we figured out that the trace is consuming a lot of performance. So the good old string.format is the best solution at the moment :-)